### PR TITLE
braces

### PR DESCRIPTION
**Pull request by sbillig**
Stage 1 of the pythonish -> rusty syntax transition.
- braces instead of colons and indents
- `else if` instead of `elif`
- no more `pass` statement

```
struct Empty {}
contract Foo {
  pub fn bar() {
    if true {
      1 + 2
      3 + 4
    } else if false {
    } else {
  }
}
```

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [x] OPTIONAL: Update [Spec](https://github.com/ethereum/fe/blob/master/docs/src/spec/index.md) if applicable
- [x] Add entry to the [release notes](https://github.com/ethereum/fe/blob/master/newsfragments/README.md) (may forgo for trivial changes)

- [x] Clean up commit history

### CODE REVIEW

File: docs/src/spec/expressions/name.md
Comment (cburgdorf) [inline docs/src/spec/expressions/name.md:11]: Not needed for this PR but we should probably switch from `python` syntax highlighting to `rust` syntax highlighting or else the snippets in the docs will look less good. I think that will also require a tweak in some script that performs the tests on top of the docs snippets.

File: docs/src/spec/lexical_structure/tokens.md
Comment (cburgdorf) [inline docs/src/spec/lexical_structure/tokens.md:11]: The paragraph on top ("significant whitespace") should be removed as well.

File: docs/validate_doc_examples.py
Comment (cburgdorf) [inline docs/validate_doc_examples.py:95]: I guess we should either move this back in `run_snippet` or remove `run_snippet`

File: newsfragments/707.feature.md
Comment (cburgdorf) [inline newsfragments/707.feature.md:2]: Just making sure `trusty` is an intended word play :D

**Review by cburgdorf — APPROVED** [review]
Looks great! Just left a few nitpicks.

File: docs/src/spec/expressions/name.md
Comment (sbillig) [reply docs/src/spec/expressions/name.md]: mdbook strips out lines that start with `#` in rust-formatted blocks, so I added a simple custom fe syntax highlighter instead.

**Comment by sbillig** [comment]
I'll merge this after #676 lands.

File: docs/src/spec/expressions/name.md
Comment (cburgdorf) [reply docs/src/spec/expressions/name.md]: Sweet

**Comment by mriise** [comment]
<3 I am a big fan of curly bois

**Pull request merged**
